examples/dwarf_die_tree.py

```diff
@@ -9,12 +9,22 @@
 #-------------------------------------------------------------------------------
 from __future__ import print_function
 import sys
+import posixpath
 
 # If pyelftools is not installed, the example can also run from the root or
 # examples/ dir of the source distribution.
 sys.path[0:0] = ['.', '..']
 
 from elftools.elf.elffile import ELFFile
+from elftools.common.py3compat import bytes2str
+
+# Replacement for DIE.get_full_path, with POSIX paths instead of host patsh
+def get_full_path_posix(die):
+    comp_dir_attr = die.attributes.get('DW_AT_comp_dir', None)
+    comp_dir = bytes2str(comp_dir_attr.value) if comp_dir_attr else ''
+    fname_attr = die.attributes.get('DW_AT_name', None)
+    fname = bytes2str(fname_attr.value) if fname_attr else ''
+    return posixpath.join(comp_dir, fname)    
```

Comment:
It's puzzling that you have to mostly duplicate `get_full_path` here. Also, I don't see `posixpath` in Python 3

What problem is this trying to solve, and can we solve it in some other way?

Reply:
The test of example scripts against the reference outputs would fail, if run on Windows, on every line where a directory name and a file name needed to be joined. The reference output was generated on a POSIX system and uses / as a separator; `get_full_path` uses `os.path.join`, which uses the separator from the host, which is \ on Windows.

One alternative I've considered was passing an optional, defaulted parameter to `get_full_path`, telling it which flavor of `path` to use (choices: host=default, auto, NT, POSIX) but this is just for the sake of the test, not because anyone complained, so I've decided to leave the API alone.

Reply:
Fair point about the CI.

I didn't consider Windows compatibility for these, hmm... Still I feel like we'd need to find a different solution here. TBH I'm surprised the tool still works on Windows at all! I certainly haven't tested it on Windows in many years and the CI doesn't run on Windows either.

I feel like we'll need a different solution here anyway - I don't want the logic of get_full_path duplicated in many places. The `pathlib` module has an `as_posix` utility - could this be useful to convert the returned result of `get_full_path`?

Reply:
The returned result, on Windows, is more often than not a Frankenmonster - something like `/home/john/myproject\mysource.cpp`, technically not a proper path either way. `Pathlib.Path()` on Windows seems to take those, though.

I will change the two snippets where I'd rewrite `get_full_path()` to use this instead, and leave the lineprogram bit alone. How's that?

As for general Windows compatibility, the unit test works as expected, the example test works once the path issue is fixed. Readelf would a trickier endeavor, since running the built-in `readelf` executable is off the table (except maybe with WSL, which I didn't investigate). There are several builds of readelf that run under Windows (Cygwin, Android NDK), but I've tried testing against the former and ran into output discrepancies right away. Didn't bother to debug those.

Reply:
Changed. `Pathlib` is not in Python2.